crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -29,17 +32,26 @@ impl FormatNodeRule<MatchCase> for FormatMatchCase {
         )?;
 
         if let Some(guard) = guard {
-            write!(
-                f,
-                [
-                    space(),
-                    text("if"),
-                    space(),
-                    maybe_parenthesize_expression(guard, item, Parenthesize::IfBreaks)
-                ]
-            )?;
+            write!(f, [space(), text("if"), space()])?;
+            if is_expression_parenthesized(guard.as_ref().into(), f.context().source()) {
+                guard.format().with_options(Parentheses::Always).fmt(f)?;
+            } else {
+                guard.format().fmt(f)?;
+            }
```

Comment:
Preserving parentheses should be the default behavior if you call `expression.format()`. 

```suggestion
           guard.format().fmt(f)?
```

Or you can use `Parenthesize::Optional` if it should use the same breaking logic as other expressions: e.g. that `case a if a == [1, 2, 3]` doesn't add parentheses and instead breaks after the `[`.

Reply:
Ok, thanks! Regarding your example, black have the same behavior where it breaks after `[` instead of parenthesizing the expression.

[Black playground](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AFCAI9dAD2IimZxl1N_WmBasjhgIpEITuW-X_EYBwH90PI2y6RtyT6F18kwTZW6zIC9DRQ8O19sGqwvib6m4JqF7O9ufUWT4Nx9AlluPRJKwEdeguCFpDra7xtQOgYPIiw6QAItgGLJUamVEQ7rzHBc5ki0iIHePEEh0Ckip1_Qz0xdBDZzN82a07Jw-xG2Y__KHnQAAABk1ZVA97JaUQABqwHDAgAAxomS6rHEZ_sCAAAAAARZWg%3D%3D)